include/common/bitdepth.rs

```diff
@@ -193,6 +204,26 @@ pub trait BitDepth: Clone + Copy {
 
     fn get_intermediate_bits(&self) -> u8;
 
+    fn cast_pixel_slice(bytes: &[u8]) -> &[Self::Pixel] {
+        let len = Self::pxstride(bytes.len());
+
+        assert!(bytes.len() % len == 0);
+        assert!(bytes.as_ptr() as usize % mem::align_of::<Self::Pixel>() == 0);
```

Comment:
We don't want to have to check this everytime we access the pixels, so it'd be better to check it once during creation.  I'd been planning to create a type for this.  This is probably fine in the meantime, though, just worst perf.

Reply:
> What were you thinking wrt making a custom type to handle it? Something like a `PixelBuf` type that wraps a `Vec<u8>` and has helpers for accessing the buffer as either a `&[u8]` or a `&[Pixel]`?

Yeah, basically.  It's a bit trickier for `Rav1dPicture::data` specifically since it supports custom allocators, but that'd be the gist.  The alignment/length stuff would all be checked during construction, and so these safe transmute casts would be no-ops other than dividing the length.